### PR TITLE
Add missing targets in the Makefile

### LINKED ISSUE #2: Add Makefile targets and horizon files
Eventually we will want to add the following Makefile targets 
```
publish-service:

publish-pattern:

register-pattern:
```

in addition to the corresponding `service.json` and `pattern.json` files

### LINKED ISSUE #3: Modify Makefile to allow DOCKER_HUB_ID to be overridden by env var
In the future we plan to have a more advanced CI/CD process and allowing the `DOCKER_HUB_ID` to be overridden by an environment variable will needed. 

We should change 
``` 
DOCKERHUB_ID:=ibmosquito
``` 
to the following
```
DOCKER_HUB_ID ?= ibmosquito
```

### PR DESCRIPTION
**Pull request by Ruchip16**
I have added the service.json and pattern.json files in addition to the missing targets in the makefile, check it out once and let me know the changes in case of any
@TheMosquito @t-fine

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,37 +1,62 @@
 # Extremely simple HTTP server that responds on port 8000 with a hello message.
 
-DOCKERHUB_ID:=ibmosquito
+DOCKER_HUB_ID ?= ibmosquito
```
Comment (t-fine) [inline Makefile:3]: @Ruchip16 with the change to `DOCKER_HUB_ID` make sure you change it everywhere `DOCKERHUB_ID` is referenced in the Makefile

File: Makefile [hunk 2]
```diff
@@ -1,37 +1,62 @@
...
 NAME:="web-hello-go"
```
Comment (t-fine) [inline Makefile]: Lets make this `SERVICE_NAME` to be consistent with your other changes as well

File: Makefile [hunk 3]
```diff
@@ -1,37 +1,62 @@
...
 test:
 	@curl -sS http://127.0.0.1:8000
 
 push:
-	docker push $(DOCKERHUB_ID)/$(NAME):$(VERSION) 
-
+	docker push $(DOCKERHUB_ID)/$(SERVICE_NAME):$(SERVICE_VERSION) 
+
+publish-service:
+	    @ARCH=$(ARCH) \
```
Comment (t-fine) [inline Makefile]: I believe the issue you were seeing has to do with tab/spaces. Makefiles can be pretty picky about that. Make sure all the lines below each target are one tab in. (I believe it's tabs it wants)

File: Makefile [hunk 4]
```diff
@@ -1,37 +1,62 @@
...
+        SERVICE_NAME="$(SERVICE_NAME)" \
+        SERVICE_VERSION="$(SERVICE_VERSION)"\
+        SERVICE_CONTAINER="$(DOCKERHUB_ID)/$(SERVICE_NAME):$(SERVICE_VERSION)" \
+        hzn exchange service publish -O $(CONTAINER_CREDS) -f service.json --pull-image
+
+publish-pattern:
+	    @ARCH=$(ARCH) \
```
Comment (t-fine) [inline Makefile]: Same goes for this target

**Review by t-fine — CHANGES_REQUESTED** [review]

**Comment by TheMosquito** [comment]
This all looks good to me, @Ruchip16.

File: Makefile [hunk 5]
```diff
@@ -1,37 +1,62 @@
...
+SERVICE_VERSION:="1.0.0"
+ARCH:="amd64"
+
+CONTAINER_CREDS:=
 
 default: build run
 
 build:
-	docker build -t $(DOCKERHUB_ID)/$(NAME):$(VERSION) .
+	docker build -t $(DOCKER_HUB_ID ?= ibmosquito)/$(SERVICE_NAME):$(SERVICE_VERSION) .
```
Comment (t-fine) [inline Makefile]: You can remove `?= ibmosquito` everywhere except for line 3. It's only needed at the top to set it initially

**Review by t-fine — CHANGES_REQUESTED** [review]

File: Makefile (hunk 5 above)
Comment (Ruchip16) [reply Makefile]: yess I had done that but not pushed it yet, I would do it asap

**Review by t-fine — APPROVED** [review]

**Pull request merged**